src/nodes/hypertable_modify.c

```diff
@@ -820,8 +820,9 @@ ExecModifyTable(PlanState *pstate)
 			bool isNull;
 
 			relkind = resultRelInfo->ri_RelationDesc->rd_rel->relkind;
-			/* Since this is a hypertable relkind should always be RELKIND_RELATION. */
-			Assert(relkind == RELKIND_RELATION);
+			/* Since this is a hypertable relkind should be RELKIND_RELATION. or
+			 * a chunk that is a foreign table (OSM chunks) */
+			Assert(relkind == RELKIND_RELATION || relkind == RELKIND_FOREIGN_TABLE);
```

Comment:
OSM chunks are still part of a hypertable so why do we need to allow foreign tables here?